### PR TITLE
Add a class method in potential to create Potential Object. Closes #358.

### LINKED ISSUE #358: Add a class method to create a Potential out of PotentialTemplate

### PR DESCRIPTION
**Pull request by umesh-timalsina**
Closes #358.

### CODE REVIEW

File: gmso/core/potential.py [hunk 1]
```diff
@@ -209,6 +209,39 @@ def __repr__(self):
...
+        potential_template : gmso.lib.potential_templates.PotentialTemplate,
+                            The potential template object
+        parameters : dict,
+                    The parameters of the potential object to create
+        topology : gmso.Topology, default=None
+                   The topology to which the created potential object belongs to
+
+        Returns
+        -------
+        gmso.Potential,
```
Comment (justinGilmer) [inline gmso/core/potential.py]: Should it be this?
```suggestion
        gmso.Potential :
```

File: gmso/tests/test_potential.py [hunk 2]
```diff
@@ -222,3 +224,13 @@ def test_set_expression_and_params_mismatch(self):
                     'u': 1.0*u.g,
                     'v': 1.0*u.m},
             )
+
```
Comment (justinGilmer) [inline gmso/tests/test_potential.py]: Would it make sense to have a test that instantiates a type of potential (harmonic bond or anything is fine) the traditional way, and then also make one with the template fashion and check if all of their attributes are the same?

File: gmso/core/potential.py (hunk 1 above)
Comment (umesh-timalsina) [reply gmso/core/potential.py]: According to numpy docstring guide, there should neither be ',' nor ':'  in the docstring for Returns in this case:
```


Explanation of the returned values and their types. Similar to the Parameters section, except the name of each return value is optional. The type of each return value is always required:

Returns
-------
int
    Description of anonymous integer return value.

If both the name and type are specified, the Returns section takes the same form as the Parameters section:

Returns
-------
err_code : int
    Non-zero value indicates error code, or zero on success.
err_msg : str or None
    Human readable error message, or None on success.


```

File: gmso/tests/test_potential.py (hunk 2 above)
Comment (umesh-timalsina) [reply gmso/tests/test_potential.py]: Ok, I will add the test.

**Pull request merged**